portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -43,8 +43,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     int randomIndex = (int)Math.floor(Math.random() * ((int)jsonString.size()));
     System.out.print(jsonString.size() + " randome num: " + randomIndex);
     response.getWriter().println(jsonString.get(randomIndex).toString());
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    boolean upperCase = Boolean.parseBoolean(getParameter(request, "upper-case", "false"));
+    
+    // Convert the text to upper case.
+    if (upperCase) {
+      text = text.toUpperCase();
+    }
 
+    //@TODO color still need to be implemented 
 
+    // Respond with the result.
+    response.setContentType("text/html;");
+    response.getWriter().println("Thanks for your comment: \n\"" + text + "\"\n Hope you have a good day!");
+    
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
```

Comment:
return value == null ? defaultValue : value;

Reply:
I improved this function and push it to  b68273de19ff9722a05935adf40f01ea1f1a3652 this commit https://github.com/Phoebe-Khokgawe/STEP_Starter/commit/b68273de19ff9722a05935adf40f01ea1f1a3652#diff-23a0cb4c16cf41219c2807c8ac14ee1d7c554d240941f51ba63f10cc057b10ba.

Reply:
nice